### ISSUE TITLE
Upload my own project trained (google colab by yolov5s) in Ultralytics Hub

### ISSUE THREAD

**Issue by leandroipca** [ultralytics/hub#43]
### Search before asking

- [X] I have searched the HUB [issues](https://github.com/ultralytics/hub/issues) and [discussions](https://github.com/ultralytics/hub/discussions) and found no similar questions.

### Question

Hey, 

I have already done some training with yolov5s to use it with the tensorflowlite app. I have some models already trained, with their weights (best.pt and respective *.tflite). Is there a way to upload the models I have already trained? They were all trained with the steps of Yolov5 from Git Hub.

Thanks

**Comment by kalenmike** [comment]
@leandroipca Thank you for raising a question. Unfortunately we do not have the option to import existing models and this featured is not currently planned for release.

Please feel free to raise any other feature requests that you believe will be useful and we will add them for consideration.

**Comment by leandroipca** [comment]
Hey,

Thank you.

The only way is to upload the images and create a model, correct?

I tried it a few times, but after uploading the model, when processing it generates an error.


thanks a lot.

**Comment by glenn-jocher** [comment]
@leandroipca hi sorry to hear that! Can you please post a screenshot of the error you are seeing and let us know your system details, i.e. browser and os type?

**Comment by kalenmike** [comment]
@leandropica It would be helpful to have some more information so we can look into your specific situation, but saying that the main cause for a dataset not processing is an incorrectly formatted dataset. Here are some things to look out for:

1. The dataset zip must contain a folder with the same name as the zip
2. [A correctly formatted YAML file](https://codebeautify.org/yaml-parser-online) must be present inside the dataset folder
3. The YAML must be configured correctly to match how your images and labels are structured, this is case sensitive
4. The labels must be in YOLOv5 format `class x_center y_center width height`

You can find an [example zip here](https://github.com/ultralytics/hub/blob/master/coco6.zip)
And you [can go here](https://github.com/ultralytics/hub/blob/master/README.md#1-create-a-dataset) to get a more detailed explanation of formatting your dataset

**Comment by leandroipca** [comment]
Hey,

It's working. The problem was in the *yaml file. I converted the same way it was in coco6. I'm already running the tests.

One question, is there a possibility to leave the model we trained for public instead of private?


best regards,

Leandro

**Comment by stefani-kovachevska** [comment]
Glad to hear that  @leandroipca , thank you for getting back to us.

> One question, is there a possibility to leave the model we trained for public instead of private?

This is a feature we're planning to develop for the following versions. If there's something else you'd like to see on Ultralytics HUB, please let us know.

Best, 
Stefani

**Comment by glenn-jocher** [comment]
@leandroipca agree we need public views ASAP. Feature is in our product roadmap.

**Comment by leandroipca** [comment]
Hey,

So, I'm testing your app because I'm using yolov5 for my MSC thesis, here in Portugal. I was testing several other types of detection algorithms like detectron2, TensorFlow, yolo (tiny, X, v3, v4(keras, pytorch and darknet)) . With yolov4 PyTorch I was able to convert the weights to tensorflowlite and test it in the tensorflow app itself (TFL Detect), however their app doesn't have the features like the Ultralytics HUB, such as calibration like IOU, confidence, ... , in addition to being able to change the dataset/model for the detection.

If it's okay for you, I can mention your app in my dissertation, and show you some pictures of how it works.

The dissertation has as its theme the help of visually impaired people in order to detect defects in garment.


thank you so much,

**Comment by glenn-jocher** [comment]
@leandroipca yes of course you can mention our app in your dissertation! 

Thank you for your feedback and thank you for your efforts to develop AI tools to help the visually impaired, this is a worthy use case.
